integrations/maya/src/main/java/org/dash/wallet/integrations/maya/di/MayaModule.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2023 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.dash.wallet.integrations.maya.di
+
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import org.dash.wallet.common.WalletDataProvider
+import org.dash.wallet.integrations.maya.api.MayaEndpoint
+import org.dash.wallet.integrations.maya.api.ExchangeRateApi
+import org.dash.wallet.integrations.maya.api.FiatExchangeRateAggregatedProvider
+import org.dash.wallet.integrations.maya.api.FiatExchangeRateProvider
+import org.dash.wallet.integrations.maya.api.MayaApi
+import org.dash.wallet.integrations.maya.api.MayaApiAggregator
+import org.dash.wallet.integrations.maya.api.RemoteDataSource
+import org.dash.wallet.integrations.maya.utils.MayaConstants
+import javax.inject.Singleton
+import kotlin.time.ExperimentalTime
+
+@Module
+@InstallIn(SingletonComponent::class)
+@ExperimentalCoroutinesApi
+@ExperimentalTime
+@FlowPreview
```

Comment:
Better to replace them with `OptIn` annotations to avoid propagating all of these up.